packages/noya-designsystem/src/components/GridView.tsx

```diff
@@ -0,0 +1,103 @@
+import { memo, useCallback, ReactNode } from 'react';
+import styled from 'styled-components';
+
+const Grid = styled.div(({ theme }) => ({
+  flex: 1,
+  color: theme.colors.text,
+  display: 'grid',
+  padding: '14px',
+  gridTemplateColumns: 'repeat(auto-fill, 200px)',
+  gridTemplateRows: 'repeat(auto-fill, 180px)', // Makes it go
+  justifyContent: 'space-between',
+  gap: '20px',
+  overflowY: 'scroll',
+}));
+
+const ScrollArea = styled.div(({ theme }) => ({
+  flex: '1 1 0px', // Ignore the grid's intrinsic height
+  display: 'flex',
+  overflowY: 'auto',
+  flexDirection: 'column',
+}));
+
+const Container = styled.div(({ theme }) => ({
+  flex: '1',
+  display: 'flex',
+  flexDirection: 'column',
+}));
+
+const ItemContainer = styled.div<{ selected: boolean }>(
+  ({ theme, selected }) => ({
+    width: '200px',
+    height: '130px',
```

Comment:
Instead of hardcoding the height here, let's make `SwatchContainer` a flex container with `display: 'flex', flexDirection: 'column'`. Then this `ItemContainer` can be `flex: 1` to fill the available space.

Reply:
Without adding the height. It looks like this.
![image](https://user-images.githubusercontent.com/16924758/111232642-81ea6c00-85b9-11eb-9388-b421987da313.png)
Is there any other way? To not have the height hardcoded. Here is the code:

```javascript
const SwatchContainer = styled.div(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  flex: '1',
}));

const ItemContainer = styled.div<{ selected: boolean }>(
  ({ theme, selected }) => ({
    display: 'flex',
    flex: '1',
    (...)
  }),
);

```

Reply:
I think we may want to use a fixed height for rows. I have never used this property before 😂  but it seems to work:

```ts
const Grid = styled.div(({ theme }) => ({
  // gridTemplateRows: 'repeat(auto-fill, 180px)'
  gridAutoRows: '180px',
  // ...
}));
```

Reply:
Minor thing: we probably don't need `flex: 1` on the SwatchContainer, since its parent is `display: grid` so it would have no effect

Reply:
Oh I see, that way the height is not in the item itself. Great!